### ISSUE TITLE
Keyboard shortcuts not working 

### ISSUE THREAD

**Issue by darkwomb** [persei802/QtDragon_hd#6]
Hello Jim! Happy new year. 

Is it possible to map keyboard buttons to specific functions in qtdragon_hd? 

I.e. a relay or cycle start etc.?

Thanks

**Comment by persei802** [comment]
Yes, that should be possible. Qtdragon_hd maps various keys to various functions defined in the handler. Examples are shown in the handler file shortly after the HandlerClass definition with statements like:
`KEYBIND.add_call('Key_F4', 'on_keycall_F4')`
To map a key to an external function like a relay, you would have to create a HAL pin and connect it to an output pin of a Mesa card. The function that the key calls would toggle the state of the HAL pin. You could indicate the state via a status message or add an LED to the user interface. 
If you are looking for a complete custom key mapping facility (like video games) that's not something I'm planning on doing.

**Comment by persei802** [comment]
I fixed the keybindings for F1 and spacebar. The others work as intended. Thanks for pointing it out.

**Comment by darkwomb** [comment]
Thank you for information! I'll played around with KEYBIND.
